skops/utils/_fixes.py

```diff
@@ -6,3 +9,74 @@ def boxplot(ax, *, tick_labels, **kwargs):
         return ax.boxplot(tick_labels=tick_labels, **kwargs)
     except TypeError:
         return ax.boxplot(labels=tick_labels, **kwargs)
+
+
+def construct_instances(estimator):
+    """Create a test instance of an estimator for compatibility testing.
+
+    This function provides compatibility between different scikit-learn versions
+    (before and after 1.6) for creating test instances of estimators. It handles
+    the API change where _construct_instances was moved from estimator_checks to
+    instance_generator.
+    """
+    try:
+        from sklearn.utils._test_common.instance_generator import _construct_instances
+
+        return next(_construct_instances(estimator))
+
+    except ImportError:
+        from sklearn.utils.estimator_checks import _construct_instance
+
+        return _construct_instance(estimator)
+
+
+def get_tags(estimator):
+    """Get estimator tags in a consistent format across different sklearn versions.
+
+    This function provides compatibility between sklearn versions before and after 1.6.
+    It returns a SimpleNamespace object containing metadata about the estimator's
+    requirements and capabilities (e.g., input types, fitting requirements).
+
+    Parameters
+    ----------
+    estimator : estimator object
+        A scikit-learn estimator instance.
+    """
+    try:
+        from sklearn.utils._tags import get_tags
+
+        tags = get_tags(estimator)
+        return SimpleNamespace(
```

Comment:
I think it's nicer to basically copy the `_old_to_new` code from sklearn, and always use the new format.

Reply:
I'm not sure, but this is what we do now:

https://github.com/scikit-learn/scikit-learn/blob/f54c98ea3b47a8b3497d6b0f0b53d20160a10240/sklearn/utils/_tags.py#L457

Reply:
I would've really appreciated to know that this is being done a bit earlier so I didn't have to reinvent the wheel on my own. Maybe using the translation code directly from sklearn is an option too? They are private functions, but in this case it might be justified?

Reply:
it was done just a few days ago by Guillaume, we weren't thinking of doing this before last week.

I don't think we should use the functions since they're to be removed in 1.7, but it makes sense to copy some of that content maybe?

Reply:
ok I will make it work, it won't be complex. check the [last issue above](https://github.com/skops-dev/skops/pull/447#discussion_r1856622611) with the loss objects and maybe undo the temp fix to see the error for yourself, as that one I had no clue what to do about. It seemed counter intuitive to add the old loss to the `trusted` again.

Reply:
I don't really understand this last message 😁

Reply:
@adrinjalali this is working in [cfeef0a](https://github.com/skops-dev/skops/pull/447/commits/cfeef0a7d3930118916d554ee459c126114f91bc) with everything except the `get_tags` function copied from that commit. It is a lot of code and it will complain about tests, and we will have to also most likely copy those. I just wanted to see if this will do, and what is the min tests required by `codecov` as this is well trialed and tested in `scikit-learn` already. It seems CI won't run the docs build, which is all good locally, maybe you can retrigger it?